compiler/src/iree/compiler/API/test/BUILD

```diff
@@ -0,0 +1,27 @@
+# Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+load("//build_tools/bazel:native_binary.bzl", "native_test")
+
+package(
+    default_visibility = ["//visibility:public"],
+    licenses = ["notice"],  # Apache 2.0
+)
+
+cc_binary(
+    name = "compile-mhlo-test-binary",
+    srcs = ["compile-mhlo-test-main.c"],
+    deps = [
+        "//compiler/src/iree/compiler/API:CAPI",
+        "//runtime/src/iree/base",
```

Comment:
Note the runtime dep - I had to turn layering check off for this. I don't see how this differs from the ConstEval usage

```
compiler/src/iree/compiler/API/test/compile-mhlo-test-main.c:15:10: error: use of private header
from outside its module: 'iree/base/string_builder.h' [-Wprivate-header]
#include <iree/base/string_builder.h>
         ^
```

(same error with `#include "iree/base/string_builder.h"`)

Reply:
Can you make the "-layering_check" explicit? Partially just to call it out and partially because it's on by default internally and so we'll have to explicitly disable it.

Reply:
we shouldn't be including that file directly - just iree/base/api.h - the layering check is yelling because the header is private